Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -457,11 +483,22 @@ extension PlaceCarouselViewController: LocationMonitorDelegate {
     func locationMonitor(_ locationMonitor: LocationMonitor, userDidExitCurrentLocation location: CLLocation) {
         locationMonitor.refreshLocation()
     }
+
+    func locationMonitor(_ locationMonitor: LocationMonitor, didFailInitialUpdateWithError error: Error) {
+        // Note: actually, at this point, if the location comes, `didUpdateLocation` will be called.
+        presentNoLocationAlert()
+    }
 }
 
 extension PlaceCarouselViewController: PlacesProviderDelegate {
     fileprivate func showErrorMessageIfNoPlaces() {
         if self.places.isEmpty {
+            let hasLocation = locationMonitor.getCurrentLocation() != nil
+            guard hasLocation else {
+                presentNoLocationAlert()
```

Comment:
You might be able to use the `presentedViewController` property of this view controller to see if it's the alert view you're showing. You'll probably have to have a reference to it to do the comparison though. Worst case, you can just use a boolean I suppose.